### DISCUSSION TITLE
Function to detect JS code nested in JSX

### DISCUSSION THREAD

**Discussion by llemaitre19** [llemaitre19/jtsx#8] [General]
> I will also look into strategies to detect JS code nested within JSX, starting with the implementation of your commenting code. 

@ramblehead , I had some free time these days to look at this point. [This commit](https://github.com/llemaitre19/jtsx/commit/c2479f0ca6f225df02a52dfe9c2e5a7dce05673b)  (+ [this one](https://github.com/llemaitre19/jtsx/commit/2804ea0ebcf991cc9ea9b5922a3e9e8821e83198)) extracts the logic for detecting JS nested in JSX of the comment feature into the  `jtsx-js-nested-in-jsx-context-p` and `jtsx-js-nested-in-jsx-context-at-p` functions.

It is on [development branch](https://github.com/llemaitre19/jtsx/tree/development) for now. I will merge it on master within few days if I do not notice any regression.

**Comment by eki3z** [comment]
@llemaitre19 There are some font-lock lacked in jtsx-jsx-mode compared to `js-jsx-mode`. Such as
In `js-jsx-mode`:
<img width="594" alt="截屏2024-05-25 21 03 58" src="https://github.com/llemaitre19/jtsx/assets/35212403/1f9bb247-dff3-44a5-9069-0caa4089dafe">

In `jtsx-jsx-mode`:
<img width="594" alt="截屏2024-05-25 21 04 21" src="https://github.com/llemaitre19/jtsx/assets/35212403/412a4efd-3b90-49f3-9034-9ad3a72c459a">

**Reply by llemaitre19** [reply]
`jtsx-jsx-mode` derives from the built-in `js-ts-mode` so if I you notice the same syntax highlighting issues when using `js-ts-mode` directly, you should consider to send a bug report to Emacs team (`M-x report-emacs-bug`).

Else, please fill an [issue](https://github.com/llemaitre19/jtsx/issues) in this repository.

**Reply by eki3z** [reply]
`js-ts-mode` has the same syntax highlighting , but it's for js file, so there is no need to highlight jsx by default. `jtsx-jsx-mode` is for jsx, I mean that need extra enhancement highlight for jsx.
